pkg/app/app.go

```diff
@@ -48,33 +49,53 @@ func App(ctx context.Context, httpClient *http.Client, configItem *config.Items)
 		return err
 	}
 
+	var wg sync.WaitGroup
+	channel := make(chan error, len(configItem.Bins))
+
+	go func() {
```

Comment:
This is probably not doing what you want it to do.  This go routine will either a) run before line 60, and finish immediately (because the count in `wg` is 0), or it will get scheduled after line 60 and it will complete after all the go routines are done.  But this only means that _this_ go routine will continue - not any other go routines.

Side note: Closing the channel here will cause your error code (lines 70-75) to fail.

Reply:
Yeah that sounds reasonable, I had a similar thought but I saw some person doing this on some blog and I thought hey they know more than me :)

Do I understand you correctly that I should remove the entire 
```
go func(){
  wg.Wait()
  close(channel)
}()
```
?

Reply:
Yes.  Remove the entire go routine entirely.